cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TemplateUtils.java

```diff
@@ -31,6 +43,61 @@ public static Map<String, String> labels(HasMetadataTemplate template)   {
         }
     }
 
+    /**
+     * Add additional volumes for a given pod template and a list of existing volumes.
+     *
+     * @param templatePod     The pod template that contains the additional volumes.
+     * @param existingVolumes The list of existing volumes to which the additional volumes will be added.
+     */
+    public static void addAdditionalVolumes(PodTemplate templatePod, List<Volume> existingVolumes) {
+        if (templatePod.getAdditionalVolumes() != null) {
+            templatePod.getAdditionalVolumes().forEach(volumeConfig -> existingVolumes.add(createVolumeFromConfig(volumeConfig)));
```

Comment:
AFAICS nothing in the operator is checking for duplicate volume names in the list. Allowing users to add volumes means duplicates are now possible. Do we want to detect and prevent that in the operator, or are we happy with whatever kube does in this situation (I don't know off hand what the kube behaviour is)?

Reply:
kubernetes should give some very meaningful error descriptions. I think it would be redundant :)